docs/developing.md

```diff
@@ -0,0 +1,153 @@
+# Developing Minishift
+
+This section describes different aspects of extending, building, and releasing Minishift.
+
+## Adding a New Dependency
```

Comment:
I guess this is the content from former _ ADD_DEPENDENCY.md _. As discussed, this can be deleted completely.